src/hera/task.py

```diff
@@ -89,6 +92,9 @@ class Task(IO):
         function call with the given parameters. Otherwise, if parallel is False and the list of params is a list of
         lists (each list contains a series of values to pass to the function), the task will execute as a task
         group and schedule multiple function calls in parallel.
+    with_sequence: Optional[dict] = None,
```

Comment:
I like this! It's super flexible. I wonder if there are some problems that some "magic" object, `Sequence`, would solve for the user?

Reply:
I could be ok with this, but then I feel like we should also add `value_from` as a dedicated object.

Reply:
Sounds reasonable. I think we can always add such magic and support `Union[dict, MagicSequence]` :)